Yosemite/Yosemite/Stores/ProductCategoryStore.swift

```diff
@@ -89,6 +91,23 @@ private extension ProductCategoryStore {
         }
     }
 
+    /// Create a new product category associated with a given Site ID.
+    ///
+    func addProductCategory(siteID: Int64, name: String, parentID: Int64?, onCompletion: @escaping (ProductCategory?, Error?) -> Void) {
```

Comment:
⚠️ We should use `Result` now that's available. See https://github.com/woocommerce/woocommerce-ios/pull/2261. 

```suggestion
    func addProductCategory(siteID: Int64, name: String, parentID: Int64?, onCompletion: @escaping (Result<ProductCategory, Error>) -> Void) {
```